force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -358,12 +444,15 @@ public with sharing class ClassesObjectsMethods {
 
         // Loop through each name in the provided list of names
             // Create a new Person object for each name
-            
             // Add the new Person object to the list of Person objects
-        
+            List<Person> newPersons = new List<Person>();
+            for (String name : names) {
+                Person newlyCreatedPerson = new Person(name, age);
+                newPersons.add(createPerson(name, age));
```

Comment:
No need to create a new person again, you've already added it to the variable on 450, you can just add your variable here.